sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -85,8 +88,13 @@ public String getId() {
         if (options == null) {
             options = new CosmosContainerRequestOptions();
         }
-        return database.getDocClientWrapper().readCollection(getLink(), ModelBridgeInternal.toRequestOptions(options))
-                   .map(response -> ModelBridgeInternal.createCosmosAsyncContainerResponse(response, database)).single();
+
+        final CosmosContainerRequestOptions requestOptions = options;
+        if(!database.getClient().getTracerProvider().isEnabled()){
+            return readInternal(options);
+        }
+
+        return withContext(context -> read(requestOptions, context)).subscriberContext(TracerProvider.callDepthAttributeFunc);
```

Comment:
this will have additional cost even if tracer not enabled.
i see in the code withContext is doing Map creation etc.

we should short circuit this if tracer is not enabled.

Reply:
its already after the tracer enable check